### PR TITLE
Connection Package: add Jetpack_Data methods

### PR DESCRIPTION
**Pull request by roccotripaldi**
Fixes nothing

#### Changes proposed in this Pull Request:

* Moves Jetpack_Data::is_usable_domain to the connection package
* Deprecates Jetpack_Data::get_access_token and moves to connection package
* Deletes Jetpack_Data:: php_bug_66229_check as it's no longer used.

Note integration unit tests for Jetpack_Data now test Connection_Manager instead.

#### Testing instructions:
- checkout this branch
- `cd packages/connection`
- `composer phpunit`

#### Proposed changelog entry for your changes:

* None

### CODE REVIEW

File: class.jetpack.php [hunk 1]
```diff
@@ -4908,8 +4912,13 @@ public static function get_secrets( $action, $user_id ) {
 		return $secrets;
 	}
 
+	/**
+     * @deprecated 7.5
```
Comment (lezama) [inline class.jetpack.php]: some indentation weirdness in this method

File: class.jetpack.php [hunk 2]
```diff
@@ -4878,6 +4878,10 @@ public static function xmlrpc_api_url() {
 		return untrailingslashit( $base ) . '/xmlrpc.php';
 	}
 
+	public static function connection() {
```
Comment (lezama) [inline class.jetpack.php]: some indentation weirdness in this method

File: class.jetpack-data.php [hunk 3]
```diff
@@ -1,226 +1,13 @@
 <?php
 
-use Automattic\Jetpack\Constants\Manager as Constants_Manager;
+use \Automattic\Jetpack\Connection\Manager as Connection_Manager;
 
 class Jetpack_Data {
```
Comment (lezama) [inline class.jetpack-data.php]: seems like it is possible to @deprecate classes, should we?

**Review by gravityrail — APPROVED** [review]
LGTM 👍

**Pull request merged**